packages/react/src/components/Popover/index.js

```diff
@@ -7,13 +7,22 @@
 
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useRef, useState } from 'react';
+import useIsomorphicEffect from '../../internal/useIsomorphicEffect';
+import { useMergedRefs } from '../../internal/useMergedRefs';
 import { usePrefix } from '../../internal/usePrefix';
 
-const Popover = React.forwardRef(function Popover(props, ref) {
+const PopoverContext = React.createContext({
+  floating: {
+    current: null,
+  },
+});
+
+const Popover = React.forwardRef(function Popover(props, forwardRef) {
   const {
     align = 'bottom',
     as: BaseComponent = 'span',
+    autoAlign = false,
```

Comment:
Did we ever document the different strategies we were looking at for auto alignment? e.g. `align="auto"` was another approach and I wasn't sure what the process was for `autoAlign`.

Are you hoping to ship this in stable next release or were you hoping to guard this behind a flag / indicate it's unstable?

Reply:
I did not document the different potential strategies. I can document what I ended up choosing  somewhere if you think that's necessary!  

 With `autoAlign` being off by default, I thought it would be okay to include it in the next release without a guard since people will have to opt in to it. I would prefer to not make it officially experimental but I'm down to put a note in the prop description for it